.github/workflows/linting.yml

```diff
@@ -0,0 +1,80 @@
+---
+name: Linting
+
+on:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - master
+    tags:
+      - 'v**'
+
+jobs:
+  code-gen:
+    name: Submariner K8s API Code Generation
```

Comment:
Now that I think of it, is this really a validation?

This generates the clientset and such, but what's the validation exactly?

Do we expect it to fail or something?

Or are we expecting git state to be clean after running it (because there's no such check currently here so it won't matter if anything changed or not)?

Reply:
I think it’s useful to make sure the code generation process works (we’ve broken it in the past).

Reply:
Good questions, I'm not sure, I've basically carried that job over into this workflow as-is.

Reply:
Yeah, I wondered that too, but I couldn't find a better category. We only want to validate that API code gen still works with our code.

Reply:
OK then let's go a step further and assert after the command ran that the git tree is clean since otherwise you'd need to manually regenerate the API

Reply:
> Good questions, I'm not sure, I've basically carried that job over into this workflow as-is.

Yeah I know just want to make sure it makes sense :)

Reply:
> OK then let's go a step further and assert after the command ran that the git tree is clean since otherwise you'd need to manually regenerate the API

Maybe you'd like to propose that in a different PR? It's not really in-scope for "reorganizing GHA jobs/workflows" I think.

Reply:
Yeah sure we can do it in a follow up to make this check more robust

Reply:
Raised https://github.com/submariner-io/submariner/issues/809 to track.